roxie/ccd/ccdfile.cpp

```diff
@@ -1786,6 +1788,14 @@ class CRoxieFileCache : implements IRoxieFileCache, implements ICopyFileProgress
                         reportedFilesToCopy = true;
                         if (iShouldCopy)
                         {
+                            if (!useRemoteResources)
+                            {
+                                b.leave();
```

Comment:
What happens if multiple threads/queries try to copy the same file at the same time?  Won't they try and copy the file multiple times?

Reply:
In theory, the copy only happens when a file is added to the cache. One thread will add to the cache and do the copy, the others should find the file already in the cache and get a (lazy, initially remote) IFileIO to it.